synapse/appservice/api.py

```diff
@@ -179,7 +178,7 @@ async def query_user(self, service: "ApplicationService", user_id: str) -> bool:
         try:
             response = await self._send_with_fallbacks(
                 service,
-                [APP_SERVICE_PREFIX, ""],
+                [APP_SERVICE_PREFIX],
```

Comment:
I vacillated here (and all other spots below) between feeling that removing the legacy prefix was enough or thinking it might make sense to stop using the `_send_with_fallbacks` mechanism entirely - open to thoughts about this.

Reply:
If there's only a single path then we should stop using `_send_with_fallbacks` which was only added for this reason.

Reply:
(I *think* we can also remove `_send_with_fallbacks` too, unless other users have cropped up!)